extensions/build-gradle/main.ts

```diff
@@ -0,0 +1,366 @@
+    if(!pkgs.length) {
+        return;
+    }
+
+    logSuccess("Submitting to Phylum for analysis...");
+        
+    if(!project) {
+        logSuccess("Please specify a project with --project <projectName>");
+    }
+
+    if(group && !project) {
+        logError("ERROR: You cannot specify a group without a project.");
+        return;
+    }
+
+    if(project) {
+        logSuccess(`\t --> Project: ${project}`)
+    }
+
+    if(group) {
+        logSuccess(`\t --> Group: ${group}`)
+    }
+
+    if(!group && !project) {
+        logError("ERROR: You must specify a project (and optionally a group).");
+        return;
+    }
+
+    if(pkgs.length) {
+        const jobId = await PhylumApi.analyze("maven", pkgs, project, group);
+        logSuccess(`Job submitted for analysis with job ID: ${jobId}`);
+    } else {
+        logSuccess("No packages to submit");
+    }
+}
+
+/**
+ *  The extension API does not currently provide facilities for creating a project.
+ *  Shim out to the Phylum CLI to handle this (inception!).
+ */
+async function attemptCreateProject(projectName: string, group?: string) {
+    projectName = projectName.replace(/[ &;:]/, "");
+    let cmd = ["phylum", "projects", "create", projectName];
+
+    if(group) {
+        group = group.replace(/[ &;:]/, "");
+        cmd = cmd.concat(["--group", group]);
+    }
+
+    // Attempts to create the project.
+    let resp = await Deno.run({
+        cmd: cmd,
+        stdout: "piped",
+        stderr: "piped",
+    }).output();
+
+    // Attempt to determine if the project was created.
+    const ret = new TextDecoder().decode(resp);
+    if(ret.indexOf("already exists") || ret.indexOf("Successfully created")) {
+        return true;
+    }
+
+    return false;
+}
+
+/**
+ *
+ */
+async function getSubprojectDependencies(rootProject?: string) {
+    if(!rootProject) {
+        logError("You must specify a root project for subprojects");
+        return;
+    }
+
+    let projects = await getGradleProjects();
+
+    let ret = {};
+
+    for(let proj in projects) {
+        let subproj = projects[proj]; 
+        const projName = `${rootProject}/${subproj.replace(":","")}`;
+        ret[projName] = await getBuildGradleDeps(subproj)
+    };
+
+    return ret;
+}
+
+/**
+ *  A cheap CLI argument parser for our extension.
+ */
+function parseArg(arg: string, cliArgs: string[]) {
+    for(let i = 0; i < cliArgs.length; i++) {
+        if(cliArgs[i] === `--${arg}`) {
+            if(i+1 < cliArgs.length) {
+                return cliArgs[i+1];
+            }
+
+            break;
+        }
+    }
+
+    return null;
+}
+
+// Parse CLI args.
+const args = Deno.args;
+
+let group = parseArg("group", args);
+let project = parseArg("project", args);
+
+// If no project name is specified, take the root directory name.
+if(!project) {
+    let ret = await Deno.run({
+            cmd: ["pwd"],
+            stdout: "piped"
+    }).output();
```

Comment:
This should just be using [Deno.cwd](https://doc.deno.land/deno/stable/~/Deno.cwd)